extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/TransformedAnnotationsBuildItem.java

```diff
@@ -14,6 +14,11 @@
 
 /**
  * Makes it possible to query transformed annotations for a given annotation target.
+ * <p>
+ * Jandex {@link org.jboss.jandex.AnnotationOverlay} in its compatible mode doesn't allow to query annotations for method
+ * parameters. Therefore, if the kind of {@link AnnotationTarget} provided in any of the methods in this class equals
+ * {@link org.jboss.jandex.AnnotationTarget.Kind#METHOD_PARAMETER}, it is automatically replaced by an annotation target of its
+ * method.
```

Comment:
Sorry for being dumb, but what is `compatibility mode` and what is the reason of this limitation?

Reply:
@Ladicek knows the gory details but in a nutshell, it's Jandex's way of keeping backward compatible behavior after introducing the annotation overlay.
It basically treats method parameters as parts of methods instead of being separate.
See https://github.com/smallrye/jandex/blob/main/core/src/main/java/org/jboss/jandex/AnnotationOverlay.java#L214-L228

Reply:
FTR this text was removed as the API now automatically filters results if the `AnnotationTarget` is a method parameter.